docs/streams/developer-guide.html

```diff
@@ -267,21 +295,532 @@
+    as the changelog stream to the resulted relation. This allows Kafka Streams to continuously update the computed value upon arrivals
+    of late records after it has already been produced to the downstream transformation operators. As for <code>KTable</code>, all its transformation
+    operations can only generate another <code>KTable</code> (though the Kafka Streams DSL does provide a special function to convert a <code>KTable</code>
+    representation into a <code>KStream</code>, which we will describe later). Nevertheless, all these transformation methods can be chained
+    together to compose a complex processor topology.
+
+    We describe these transformation operations in the following subsections, categorizing them into two categories:
+    stateless and stateful transformations.
+
+   
+
+    <h5><a id="streams_dsl_transformations_stateless" href="#streams_dsl_transformations_stateless">Stateless transformations</a></h5>
+    Stateless transformations, by definition, do not depend on any state for processing, and hence implementation-wise they do not
+    require a state store associated with the stream processor.
+
+
+<table class="data-table" border="1">
+    <tbody><tr>
+        <th>Transformation</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td><b>Branch</b>: <code>KStream->KStream<</code></td>
+        <td>
+            Branch (or split) a <code>KStream</code> based on the supplied predicates into one or more <code>KStream</code> instances.<p>
+
+            Predicates are evaluated in order. A record is placed to one and only one output stream on the first match:
+            if the n-th predicate evaluates to true, the record is placed to n-th stream. If no predicate matches,
+            the record is dropped.<p>
+
+            Branching is useful, for example, to route records to different downstream topics.
+
+            <pre class="brush: java;">
+            KStream&lt;String, Long&gt; stream = ...;
+            KStream&lt;String, Long&gt;[] branches = stream.branch(
+		            (key, value) -> key.startsWith("A"), /* first predicate  */
+		            (key, value) -> key.startsWith("B"), /* second predicate */
+		            (key, value) -> true                 /* third predicate  */
+            );
+            // KStream branches[0] contains all records whose keys start with "A"
+            // KStream branches[1] contains all records whose keys start with "B"
+            // KStream branches[2] contains all other records
+            // Java 7 example: cf. `filter` for how to create `Predicate` instances
+	        </pre>
+        </td>
+    </tr>
+	 <tr>
+	   <td><b>Filter</b>: <code>KStream->KStream or KTable->KTable</code></td>
+	   <td>
+	     Evaluates a boolean function for each element and retains those for which the function returns true. 
```

Comment:
Can we link to the Javadocs here? (same comment for all these documented methods)